libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -222,6 +222,47 @@ public void checkNetworkAccess(Class<?> callerClass, int actions) {
         );
     }
 
+    public void checkWriteProperty(Class<?> callerClass, String property) {
+        var requestingClass = requestingClass(callerClass);
+        if (isTriviallyAllowed(requestingClass)) {
+            return;
+        }
+
+        // FIXME: Having to pass the entitlement class here for logging purposes seems wrong.
+        // This API suggests it's doing something else than what it actually does. This looks like a filter, but isn't!
+        ModuleEntitlements entitlements = getEntitlements(requestingClass, Entitlement.class);
```

Comment:
@ldematte this feels a bit strange, in this case I'm interested in both `WriteAllPropertiesEntitlement` and `WritePropertiesEntitlement`

Reply:
You are correct, I added this and fell into the same trap :) 
The entitlement type is present in the exception anyway, so this is redundant and also may be a bit misleading if we are looking for 2 types like you and I need to do. I'll remove it as part of https://github.com/elastic/elasticsearch/pull/120391

Reply:
Done, but feel free to merge first and I'll manage conflicts (if my PR does not go in first)